tamr_client/dataset/dataset.py

```diff
@@ -37,6 +38,9 @@ class Dataset:
     description: Optional[str] = None
 
 
+AllDataset = Union[Dataset, UnifiedDataset]
```

Comment:
To be consistent with the vernacular of the built-in `typing` module, we should name the union "AnyDataset":

```suggestion
AnyDataset = Union[Dataset, UnifiedDataset]
```